### PR TITLE
Compatibility 4.3 + 4.6

### PR DESCRIPTION
**Pull request by Stereck**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I updated the CHANGELOG.md

### CODE REVIEW

File: src/shopgate/gambiogx/includes/modules/payment/ShopgateInstallHelper.php [hunk 1]
```diff
@@ -416,4 +444,9 @@ private function sendPostRequest($data)
 
         return true;
     }
+
+    private function useLegacyConfigTable()
+    {
+        return !(TABLE_CONFIGURATION === 'gx_configurations');
```
Comment (Rekhyt) [inline src/shopgate/gambiogx/includes/modules/payment/ShopgateInstallHelper.php]:
```suggestion
        return TABLE_CONFIGURATION !== 'gx_configurations';
```

File: src/shopgate/gambiogx/includes/modules/payment/ShopgateInstallHelper.php [hunk 2]
```diff
@@ -160,9 +168,16 @@ private function getUid()
         if (file_put_contents($hashFile, $content) === false) { /*error*/
         }
 
-        $updateKeyQuery = 'UPDATE ' . TABLE_CONFIGURATION . ' AS c  SET c.configuration_value ="' . $saltedHash
-            . '" WHERE c.configuration_key = "'
-            . self::SHOPGATE_CALLBACK_DATABASE_CONFIG_KEY . '";';
+        if (!(TABLE_CONFIGURATION === 'gx_configurations')) {
```
Comment (Rekhyt) [inline src/shopgate/gambiogx/includes/modules/payment/ShopgateInstallHelper.php]:
```suggestion
        if ($this->useLegacyConfigTable()) {
```

File: src/shopgate/gambiogx/includes/modules/payment/shopgate.php [hunk 3]
```diff
@@ -568,4 +607,9 @@ private function tableExist($table)
 
         return false;
     }
+
+    private function useLegacyConfigTable()
+    {
+        return !(TABLE_CONFIGURATION === 'gx_configurations');
```
Comment (Rekhyt) [inline src/shopgate/gambiogx/includes/modules/payment/shopgate.php]:
```suggestion
        return TABLE_CONFIGURATION !== 'gx_configurations';
```

File: src/shopgate/model/location/ShopgateShippingModel.php [hunk 4]
```diff
@@ -60,4 +71,9 @@ public function getShippingConfigurationValuesByClassName($className)
 
         return $shippingConfig;
     }
+
+    private function useLegacyConfigTable()
+    {
+        return !(TABLE_CONFIGURATION === 'gx_configurations');
```
Comment (Rekhyt) [inline src/shopgate/model/location/ShopgateShippingModel.php]:
```suggestion
        return TABLE_CONFIGURATION !== 'gx_configurations';
```

**Review by Rekhyt — CHANGES_REQUESTED** [review]
Looks pretty straight forward. Only small code-style suggestions so far. Will test in a Gambio shop tomorrow.

Thank you for the PR & update.

**Comment by Stereck** [comment]
@Rekhyt thank you for the review and suggestions. Unfortunately I am missing the rights to apply them. Maybe because its a Shopgate repo or because the branch + commit is from Rainer. If you can apply them from here, can you please do so?
Otherwise just let me know and I will do it on my local and push it as suggested.

**Comment by Rekhyt** [comment]
Seems like you didn't have access anymore. I applied the suggestions and fixed permissions for next time.

File: src/shopgate/gambiogx/includes/modules/payment/ShopgateInstallHelper.php [hunk 5]
```diff
@@ -119,19 +119,27 @@ private function getUid()
...
 
-        $tokenQuery = 'SELECT c.gm_value as token FROM gm_configuration as c WHERE c.gm_key = "SECURE_TOKEN" LIMIT 1;';
-        $result     = xtc_db_query($tokenQuery);
-        $row        = xtc_db_fetch_array($result);
+        try {
+            $tokenQuery = 'SELECT c.gm_value as token FROM gm_configuration as c WHERE c.gm_key = "SECURE_TOKEN" LIMIT 1;';
+            $result     = xtc_db_query($tokenQuery);
+            $row        = xtc_db_fetch_array($result);
+        } catch (Exception $e) {
+            $row = [];
```
Comment (Rekhyt) [inline src/shopgate/gambiogx/includes/modules/payment/ShopgateInstallHelper.php]: As per PHP CS Fixer:

```suggestion
            $row = array();
```

**Review by Rekhyt — CHANGES_REQUESTED** [review]

**Review by Rekhyt — APPROVED** [review]

**Comment by Stereck** [comment]
@Rekhyt thank you! Not sure if permissions have been updated, but I cannot merge this PR ^^

**Comment by Rekhyt** [comment]
I was going to merge this anyway but good hint, permissions still weren't sufficient. Should be good now. If we ever need to touch this again.

**Pull request merged**